File: support_tools/swigwin-3.0.12/Examples/test-suite/go/template_typedef_cplx4_runme.go
package main

import . "./template_typedef_cplx4"

func main() {
	// this is OK

	s := NewSin()
	s.Get_base_value()
	s.Get_value()
	s.Get_arith_value()
	My_func_r(s)
	Make_Multiplies_double_double_double_double(s, s)

	z := NewCSin()
	z.Get_base_value()
	z.Get_value()
	z.Get_arith_value()
	My_func_c(z)
	Make_Multiplies_complex_complex_complex_complex(z, z)

	// Here we fail
	d := Make_Identity_double()
	My_func_r(d)

	c := Make_Identity_complex()
	My_func_c(c)
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/python/template_typedef_cplx4_runme.py
import string
from template_typedef_cplx4 import *

#
# this is OK
#


s = Sin()
s.get_base_value()
s.get_value()
s.get_arith_value()
my_func_r(s)
make_Multiplies_double_double_double_double(s, s)

z = CSin()
z.get_base_value()
z.get_value()
z.get_arith_value()
my_func_c(z)
make_Multiplies_complex_complex_complex_complex(z, z)

#
# Here we fail
#
d = make_Identity_double()
my_func_r(d)

c = make_Identity_complex()
my_func_c(c)
